discovery/gossiper.go

```diff
@@ -2439,6 +2452,31 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
 		return nil, false
 	}
 
+	// If this peer is banned and the channel announcement is for a channel
+	// that we are not a part of, ignore the announcement.
+	if nMsg.isRemote && d.isBanned(nMsg.peer.PubKey()) {
+		chanPeer, err := d.isChannelPeer(nMsg.peer.IdentityKey())
+		if err != nil {
+			log.Errorf("failed to check if peer %x is a channel "+
+				"peer: %v", nMsg.peer.PubKey(), err)
+			nMsg.err <- err
+
+			return nil, false
+		}
+
+		if !chanPeer {
+			nMsg.peer.Disconnect(ErrPeerBanned)
```

Comment:
Will we prevent incoming connections from being fully accepted (either at the brontide connection handshake layer, or in the server before we finalize the peer)? Otherwise, we could have a situation where they: connect, send something bad, and we disconnect (in a loop).

Reply:
Made the change in the server, but still need to properly test in a few different scenarios